src/tiled/tilesetwangsetmodel.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * tilesetwangsetmodel.h
+ * Copyright 2017, Benjamin Trotter <bdtrotte@ucsc.edu>
+ * This file is part of libtiled.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *    1. Redistributions of source code must retain the above copyright notice,
+ *       this list of conditions and the following disclaimer.
+ *
+ *    2. Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE CONTRIBUTORS ``AS IS'' AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO
+ * EVENT SHALL THE CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+ * OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+ * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <QAbstractItemModel>
+
+namespace Tiled {
+
+class Tileset;
+class WangSet;
+
+namespace Internal {
+
+class TilesetDocument;
+
+class TilesetWangSetModel : public QAbstractListModel
+{
+    Q_OBJECT
+
+public:
+    enum UserRoles {
+        WangSetRole = Qt::UserRole
+    };
+
+    TilesetWangSetModel(TilesetDocument *mapDocument,
```

Comment:
This constructor should be marked `explicit`, to avoid potential implicit conversion from `TilesetDocument *` to `TilesetWangSetModel` (however unlikely to run into).